src/cryfs-cli/program_options/ProgramOptions.cpp

```diff
@@ -16,6 +16,20 @@ ProgramOptions::ProgramOptions(const bf::path &baseDir, const bf::path &mountDir
     :_baseDir(baseDir), _mountDir(mountDir), _configFile(configFile), _foreground(foreground),
      _cipher(cipher), _blocksizeBytes(blocksizeBytes), _unmountAfterIdleMinutes(unmountAfterIdleMinutes),
      _logFile(logFile), _fuseOptions(fuseOptions) {
+
+    auto hasNoOption = [&](const char *opt) {
+        for (const string& it : _fuseOptions) {
+            if (std::strncmp(it.c_str(), opt, std::strlen(opt))) {
+                return false;
+            }
+        }
+        return true;
+    };
+
+    if (hasNoOption("subtype=cryfs") && hasNoOption("fsname=cryfs@")) {
+               _fuseOptions.push_back("-ofsname=cryfs@"+baseDir.native());
+               _fuseOptions.push_back("-osubtype=cryfs");
+    }
```

Comment:
Hm the Fuse class belongs to fspp and should be independent of the actual file system implementation. And since there might be file systems not having base directories, that information shouldn't be in the Fuse class. Sorry, I didn't notice before that we need it here.

Another option might be to pass in two string arguments to the Fuse constructor, one for the file system type (i.e. "-osubtype") and one for the file system source (i.e. "-ofsname"). Fuse has a bit strange naming here, but the man page "man fuse" is clear on what the options mean:

      fsname=NAME
              Sets the filesystem source (first field in /etc/mtab). The default is the mount program name.

      subtype=TYPE
              Sets the filesystem type (third field in /etc/mtab). The default is the mount program name. If the kernel suppports it, /etc/mtab and /proc/mounts will show the filesystem type as fuse.TYPE

              If the kernel doesn't support subtypes, the source filed will be TYPE#NAME, or if fsname option is not specified, just TYPE.

The file system type could then also be used as executable name in your other commit https://github.com/cryfs/cryfs/pull/61#discussion-diff-65478447R238